soap/wsdl-loader.js

```diff
@@ -0,0 +1,103 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: generator-loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+var url = require('url');
+var chalk = require('chalk');
+
+var fs = require('fs');
+var async = require('async');
+var soapGenerator = require('loopback-soap');
+var soap = require('strong-soap').soap;
+var WSDL = soap.WSDL;
+var path = require('path');
+
+// loads remote WSDL or local WSDL using strong-soap module APIs.
+function loadWsdl(wsdlUrl, log, cb) {
+  WSDL.open(wsdlUrl, {}, function(err, wsdl) {
+    if (err) {
+      return cb(err);
+    }
+    console.log('loadWsdl loadWsdl %j', wsdl);
+    cb(null, wsdl);
+  });
+}
+
+// get services defined in the wsdl
+exports.getServices = function getServices(wsdlUrl, log, cb) {
+  loadWsdl(wsdlUrl, log, function(err, wsdl) {
+    console.log('getServices loadWsdl %j', wsdl);
+    console.log('getServices err %j', err);
+    if (err) {
+      return cb(err);
+    }
+    this.wsdl = wsdl;
+    console.log('getServices this.wsdl %j', this.wsdl);
+    this.wsdlUrl = wsdlUrl;
+    this.services = this.wsdl.services;
+    console.log('getServices this.services  %j', this.services);
+    console.log('wsdl.definitions.services  %j', wsdl.definitions.services);
+    return cb(null,  this.services);
+  }.bind(this));
```

Comment:
See 
- http://stackoverflow.com/questions/22770299/meaning-of-this-in-node-js-modules-and-functions
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this